src/studio/src/designer/backend/Services/Interfaces/IAltinnGitRepositoryFactory.cs

```diff
@@ -0,0 +1,19 @@
+using Altinn.Studio.Designer.Infrastructure.GitRepository;
+
+namespace Altinn.Studio.Designer.Services.Interfaces
+{
+    /// <summary>
+    /// Interface for handling the creation of AltinnGitRepository instances.
+    /// </summary>
+    public interface IAltinnGitRepositoryFactory
+    {
+        /// <summary>
+        /// Creates an instance of <see cref="IAltinnGitRepository"/>
```

Comment:
Should the name of the function be create if a new instance is created with the call?

Reply:
I've looke at few descriptions of the pattern and it seems to Get that's used, this for example: https://www.c-sharpcorner.com/article/factory-method-design-pattern-in-c-sharp/